src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/DotNetCoreProjectCompatibilityDetector.cs

```diff
@@ -254,28 +286,150 @@ public int OnAfterBackgroundSolutionLoadComplete()
+        {
+            try
+            {
+                // Do we meed to update our cached data?
+                if (_curVersionCompatibilityData == null || _timeCurVersionDataLastUpdatedUtc.AddHours(CacheFileValidHours).IsLaterThan(DateTime.UtcNow))
+                {
+                    string downLoadedVersionData = null;
+
+                    // First try the cache
+                    Dictionary<Version, VersionCompatibilityData> versionCompatData = GetCompabilityDataFromCacheFile(checkTimeStamp: true);
+
+                    if (versionCompatData == null)
+                    {
+                        // Try downloading it 
+                        try
+                        {
+                            var httpClient = new DefaultHttpClient();
+                            downLoadedVersionData = await httpClient.GetStringAsync(new Uri(VersionCompatibilityFwlink)).ConfigureAwait(false);
+                            versionCompatData = VersionCompatibilityData.DeserializeVersionData(downLoadedVersionData);
+                        }
+                        catch
+                        {
+                            // If we have a cached file use it
+                            versionCompatData = GetCompabilityDataFromCacheFile(checkTimeStamp: false);
+                        }
+                    }
+
+                    if (versionCompatData != null)
+                    {
+                        // Cache the data locally
+                        if (downLoadedVersionData != null)
+                        {
+                            FileSystem.Value.WriteAllText(VersionDataCacheFile, downLoadedVersionData);
+                        }
+
+                        // First try to match exatly on our VS version and if that fails, match on just major, minor
+                        if (versionCompatData.TryGetValue(OurVSVersion, out VersionCompatibilityData compatData) || versionCompatData.TryGetValue(new Version(OurVSVersion.Major, OurVSVersion.Minor), out compatData))
+                        {
+
+                            // Now fix up missing data
+                            if (string.IsNullOrEmpty(compatData.OpenSupportedMessage))
+                            {
+                                compatData.OpenSupportedMessage = VSResources.PartialSupportedDotNetCoreProject;
+                            }
+
+                            if (string.IsNullOrEmpty(compatData.OpenUnsupportedMessage))
+                            {
+                                compatData.OpenUnsupportedMessage = string.Format(VSResources.NotSupportedDotNetCoreProject, s_partialSupportedVersion.Major, s_partialSupportedVersion.Minor);
+                            }
+                            _curVersionCompatibilityData = compatData;
+                        }
+                    }
+                }
+            }
+            catch
+            {
+
```

Comment:
Consider: Logging telemetry so that we know how often/why this ends up happening.

Reply:
The missing strings is a supported scenario where we will just use the strings (localized) that are in the product. No reason to add telemetry for this.

Reply:
I guess i meant if the download fails, or if we can't read the file from disk for some reason, so maybe the try/catch around line 352 would be a better place for telemetry.